enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,350 @@
+
+### Non-Goals
+
+There are a number of possible next-steps that I'm punting on for now, as I try to nibble off the smallest possible slice of progress.
+This enhancement will not:
+
+* Figure out how to configure a ClusterImagePolicy for verifying a mirrored release image, although this will need to happen before we can use Sigstore signatures in disconnected/restricted-network environments.
+* Suggest changes to mirroring tools, although they will need to learn to mirror Sigstore signatures for use in disconnected/restricted-network environments.
+* Suggest changes to the ImageStream controller, although it will need to learn to mirror Sigstore signatures for verification when running pods from ImageStream sources, e.g. [must-gather][].
+* Suggest policies for `quay.io/openshift-release-dev/ocp-v4.0-art-dev`, although those would help tighten down security for other `openshift-*` namespaces.
+* Support testing unsigned releases as HyperShift hosted clusters on a management cluster that requires signed releases.
+
+## Proposal
+
+This enhancement proposes a new `openshift` ClusterImagePolicy that requires Red Hat signatures for `quay.io/openshift-release-dev/ocp-release` images.
+The new ClusterImagePolicy will be managed by the ClusterVersion operator and applied to all clusters by default (initially tech-preview, gated by the `ImagePolicy` feature gate ([currently `TechPreviewNoUpgrade`][feature-gate]).
+
+Internal test clusters, which need to run releases that are not signed by official Red Hat keys, will us a new install-config property `internalTestingImagePolicy: Disable` (also gated by the `ImagePolicy` feature gate, [precedent for a feature-gated install-config property][install-config-feature-gate-precedent]).
+When set `Disable`, [the installer's ClusterVersion `overrides` handling][installer-overrides] will include an entry for the new `openshift` ClusterImagePolicy.
+When [ClusterVersion `overrides`](/dev-guide/cluster-version-operator/dev/clusterversion.md#setting-objects-unmanaged) is set from cluster-installation like this, the cluster-version operator will never create the resource, so there is no need for further tooling to delete it.
+Having `overrides` set blocks the cluster-version operator from updating, but because all internal updates between unsigned releases use `force`, that is not a problem for that use-case.
+
+### Workflow Description
+
+There is no supported workflow for this enhancement; when the relevant feature gate is enabled, this functionality works transparently without cluster administrator involvement.
+
+For Red-Hat-internal CI and QE, the agent configuring the install-config needs to set `internalTestingImagePolicy: Disable`.
+The new property can be set regardless of version or whether the `ImagePolicy` feature gate is enabled, because [the installer does not fail on unrecognized properties][install-config-unrecognized-properties].
+
+### API Extensions
```

Comment:
Status information for ClusterVersion / HostedCluster / NodePool when payload is not signed / Pod can't be deployed?

Reply:
The CVO-launches a Job to pull in the new target's manifests, and that job can have problems already.  [OCPBUGS-31462][1] is a recent example.  It's  reported via [the `ReleaseAccepted` condition][2].  Likely has no analog for HostedCluster or NodePool, because I doubt they do the warm manifest handoff that the CVO needs to do to handle [implicitly-enabled capabilities][3].  But they will likely need something similar if  they ever get around to supporting configurable cluster capabilities.  And they should already have something similar, because "the pod I tried to launch came up healthy" is not something you want to rely on happening _every_ time.

[1]: https://issues.redhat.com/browse/OCPBUGS-31462
[2]: https://github.com/openshift/cluster-version-operator/blob/5d39f27c9f645c68d0cad059dc781bad7506f310/pkg/cvo/status.go#L167-L169
[3]: https://github.com/openshift/enhancements/blob/4668a0825c59739dfafd2ae661c16cf30f540946/enhancements/installer/component-selection.md#updates

Reply:
I've added a *Existing OpenPGP user experience* section covering `ReleaseAccepted` in 1b7f6dbc -> f2d51949.